### PR TITLE
fix drink counting

### PR DESCRIPTION
**Pull request by murtagh27**

### CODE REVIEW

File: apps/frontend/src/app/events/new/page.tsx [hunk 1]
```diff
@@ -2,29 +2,62 @@
...
+          removeDrink(drink);
+        }}
+      >
+        -
+      </button>
+    </li>
+  ));
+  if (elements.length === 0) {
+    elements = <li className='text-gray-500'>Nincs hozzáadott ital</li>;
+  }
```
Comment (Tschonti) [inline apps/frontend/src/app/events/new/page.tsx]: you could put this in the return statement

```html
{consumedDrinks.size === 0 ? (
 <li>...</li>
) : (
  Array.from(consumedDrinks.entries()).map(([drink, value]) => ( ... ))
)
```

**Review by Tschonti — COMMENTED** [review]
nice work, but we have to think one thing over:
The user is creating drinkActions here, and drinkActions have price and millilitre properties (the drink entity itself contains neither of these infos (currently)). So selecting two beers doesn't make sense right now, instead the user should be able to specify the amount consumed in millilitres. 
Or we could change how we store drinks, and store a defaultVolume with each, then this could work

File: apps/backend/package.json [hunk 2]
```diff
@@ -30,12 +31,17 @@
...
+    "js-cookie": "^3.0.5",
     "nestjs-prisma": "^0.23.0",
     "passport": "^0.7.0",
     "passport-jwt": "^4.0.1",
+    "react-hook-form": "^7.53.0",
     "reflect-metadata": "^0.2.2",
     "rimraf": "^5.0.5",
-    "rxjs": "^7.8.1"
+    "rxjs": "^7.8.1",
+    "zod": "^3.23.8"
```
Comment (Tschonti) [inline apps/backend/package.json]: all these packages (the ones you added in this PR) are used by the frontend, so these should go to packages/frontend/package.json You can just remove them from this file and move it to the one in the frontend dir, and then execute yarn install

File: apps/backend/src/events/events.service.ts [hunk 3]
```diff
@@ -15,18 +15,23 @@ export class EventsService {
...
+          drinkActions:
+            drinkActions.length > 0
+              ? {
+                  create: drinkActions.map(({ drinkId, ...action }) => ({
+                    ...action,
+                    drink: { connect: { id: drinkId } },
+                    user: { connect: { authSchId: userId } },
+                  })),
+                }
+              : undefined,
```
Comment (Tschonti) [inline apps/backend/src/events/events.service.ts]: you can revert these changes, for me it works with the old version

File: apps/frontend/src/app/events/new/page.tsx [hunk 4]
```diff
@@ -13,165 +55,211 @@ const mockDrinks: simpleDrink[] = [
...
-      setConsumedDrinks((prevState) => new Map(prevState.set(drink, prevDrinkCount + 1)));
+    const existingDrinks = form.getValues('drinkActions') || [];
+    const drinkIndex = existingDrinks.findIndex((d) => d.id === drink.id);
+
+    if (drinkIndex === -1) {
+      existingDrinks.push({ ...drink });
     } else {
-      setConsumedDrinks((prevState) => new Map(prevState.set(drink, 1)));
+      existingDrinks[drinkIndex].price = drink.price;
     }
```
Comment (Tschonti) [inline apps/frontend/src/app/events/new/page.tsx]: not sure what you're trying to do here, but nothing happens if you add a drink that is already in the drinkActions array.
but the whole ordeal with the drinkactions can be handled in a future PR

File: apps/frontend/src/app/events/new/page.tsx [hunk 5]
```diff
@@ -1,10 +1,52 @@
...
+      z.object({
+        id: z.number(),
+        name: z.string(),
+        price: z.number().optional(),
+      })
+    )
+    .optional(),
+});
+
+async function onSubmit(values: z.infer<typeof formSchema>) {
```
Comment (Tschonti) [inline apps/frontend/src/app/events/new/page.tsx]: we should call `event.preventDefault()` at the begining of this function, so the standard HTML form is not submitted (you can see the URL change when you click save). Not sure how to get access to the event object in this function tho

File: package-lock.json
Comment (Tschonti) [inline package-lock.json:1]: please delete this file and remember to install packages with yarn

**Review by Tschonti — CHANGES_REQUESTED** [review]
I tried running on my machine, and event creation works without problems, as long as you don't send any drinkactions. So maybe it was just a migration error all along.

I think it's best to leave the stuff with the drinkAction dialog for later, we can merge this once the other small issues have been addressed

File: apps/frontend/src/app/events/new/page.tsx (hunk 4 above)
Comment (murtagh27) [reply apps/frontend/src/app/events/new/page.tsx]: oh it used to work, i must have messed it up.. but okay i will handle it in a future pr

**Comment by Tschonti** [comment]
just remove the console logs and vercel will be happy as well

**Review by Tschonti — APPROVED** [review]

**Pull request merged**